src/gui.rs

```diff
@@ -394,195 +375,26 @@ impl Sam {
-                });
-
-                ui.add_space(20.0);
-
-                let changed = *self.data_source.simulation_settings().unwrap() != old_settings;
-                let released = false;
-                ui.horizontal(|ui| {
-                    if ui.button("Reset").clicked() {
-                        *(self.data_source.simulation_settings().unwrap()) = SimulationSettings::default();
-                    }
-
-                    if ui.button("↻  Rerun").clicked() || (changed && released) {
-                        self.data_source.reset();
-                    }
-                });
-            });
+            create_simulation_panel(self, ctx);
         }
 
         // Top panel containing text indicators and flight mode buttons
-        if ctx.screen_rect().width() > 1000.0 {
-            egui::TopBottomPanel::top("topbar").min_height(60.0).max_height(60.0).show(ctx, |ui| {
-                ui.set_enabled(!self.archive_window_open);
-                ui.horizontal_centered(|ui| {
-                    self.top_bar(ui, false);
-                });
-            });
-        } else {
-            egui::TopBottomPanel::top("topbar").min_height(20.0).max_height(300.0).show(ctx, |ui| {
-                ui.set_enabled(!self.archive_window_open);
-                CollapsingHeader::new("Status & Controls").default_open(false).show(ui, |ui| {
-                    self.top_bar(ui, true);
-                    ui.add_space(10.0);
-                });
-            });
-        }
+        create_header(self, ctx);
 
         // Bottom status bar
-        egui::TopBottomPanel::bottom("bottombar").min_height(30.0).show(ctx, |ui| {
-            ui.set_enabled(!self.archive_window_open);
-            ui.horizontal_centered(|ui| {
-                // Give the data source some space on the left ...
-                ui.horizontal_centered(|ui| {
-                    ui.set_width(ui.available_width() / 2.0);
-                    self.data_source.status_bar_ui(ui);
-                });
-
-                // ... and the current tab some space on the right.
-                ui.allocate_ui_with_layout(ui.available_size(), Layout::right_to_left(Align::Center), |ui| {
-                    #[cfg(not(target_arch = "wasm32"))]
-                    ui.add_enabled_ui(!self.data_source.is_log_file(), |ui| {
-                        if ui.button("⏮  Reset").clicked() {
-                            self.data_source.reset();
-                        }
-                    });
-
-                    #[cfg(not(target_arch = "wasm32"))]
-                    ui.separator();
-
-                    match self.tab {
-                        GuiTab::Launch => {}
-                        GuiTab::Plot => self.plot_tab.bottom_bar_ui(ui, self.data_source.as_mut()),
-                        GuiTab::Configure => {}
-                    }
-                });
-            });
-        });
+        create_bottom_status_bar(self, ctx);
```

Comment:
I was planning to either get rid of this reset button altogether or move it to the data source specific UI, for resetting during long serial sessions. This way, there would be no actual GUI code here, just some generic GUI for the data source and current tab. In that case, I think this could be compact enough to stay here. This way all the tab-dependent stuff would happen here in `src/gui.rs` (except maybe for the buttons for selecting one, but that could be changed) and if you added a new tab you would only have to add your new tab to src/gui.rs.